src/util/cron.c

```diff
@@ -177,15 +214,45 @@ void Cron_Stop(void) {
 	cron = NULL;
 }
 
-void Cron_AddTask(uint when, CronTaskCB cb, void *pdata) {
+CronTaskHandle Cron_AddTask(uint when, CronTaskCB cb, void *pdata) {
 	ASSERT(cron != NULL);
-	ASSERT(cb != NULL);
+	ASSERT(cb   != NULL);
 
 	CRON_TASK *task = rm_malloc(sizeof(CRON_TASK));
 	task->cb     =  cb;
 	task->pdata  =  pdata;
 	task->due    =  due_in_ms(when);
+	task->state  =  TASK_PENDING;
 
 	CRON_InsertTask(task);
+
+	return (uintptr_t)task;
```

Comment:
I don't understand the purpose of this cast - why do `CRON_TASK` and `CronTaskHandle` need to be interchangable?

Reply:
I don't want to revel `CRON_TASK` outside of this file, this type is internal